src/gcode-machine-control.cc

```diff
@@ -450,6 +451,22 @@ bool GCodeMachineControl::Impl::check_for_pause() {
   return hardware_mapping_->TestPauseSwitch();
 }
 
+void GCodeMachineControl::Impl::set_lookahead(int size) {
+  // It is necessary to have a fully stopped machine to change lookahead.
+  // This is to avoid introducing a delay between the requested change and
+  // actual machine behavior with previously enqueued segments.
+  motor_ops_->WaitQueueEmpty();
+  const bool set_max = size < 0;
+  size = (size <= 0) ? -1 : size;
+  const bool set = planner_->SetLookAhead(&size);
+  if (set_max) {
+    planner_->SetLookAhead(&size);
+  }
+  if (!set) {
+    mprintf("// ERROR: allowed range 1 <= S <= %d\n", size);
```

Comment:
This is ok for now, I think in the future we need to re-think `gcode_block_done()` that it also returns an error status. So nothing to do in this PR.